sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -376,6 +376,10 @@ private[hive] class HiveClientImpl(
         unsupportedFeatures += "bucketing"
       }
 
+      if (h.getTableType == HiveTableType.VIRTUAL_VIEW && partCols.nonEmpty) {
+        unsupportedFeatures += "partitioned view"
```

Comment:
After digging it deeper and deeper, I am really doubting the initial motivation of partitioned views makes sense... 

First, see the Hive design link: https://cwiki.apache.org/confluence/display/Hive/ViewDev
> Update 30-Dec-2009: Prasad pointed out that even without supporting materialized views, it may be necessary to provide users with metadata about data dependencies between views and underlying table partitions so that users can avoid seeing inconsistent results during the window when not all partitions have been refreshed with the latest data. One option is to attempt to derive this information automatically (using an overconservative guess in cases where the dependency analysis can't be made smart enough); another is to allow view creators to declare the dependency rules in some fashion as part of the view definition. Based on a design review meeting, we will probably go with the automatic analysis approach once dependency tracking is implemented. The analysis will be performed on-demand, perhaps as part of describing the view or submitting a query job against it. Until this becomes available, users may be able to do their own analysis either via empirical lineage tools or via view->table dependency tracking metadata once it is implemented. See HIVE-1079.
> Update 1-Feb-2011: For the latest on this, see PartitionedViews.

Basically, this feature just affects the metadata of views. It does not affect the query execution.

To add the partition info into the views, users have to manually issue the SQL:
```SQL
ALTER VIEW view_name ADD [IF NOT EXISTS] partition_spec partition_spec ...
ALTER VIEW view_name DROP [IF EXISTS] partition_spec, partition_spec, ...
```

I read the code changes and test cases in the Hive JIRA: https://issues.apache.org/jira/browse/HIVE-1079. I think we do not need to worry about this Hive-specific feature. The usage scenario is very limited. Maybe the code changes in the existing PR is enough.

If you think we should support it, we might need to also need code changes in `SHOW PARTITIONS` and [`DESC table PARTITONS`](https://github.com/apache/spark/pull/15168). Then, we need to change the [`fromHivePartition` function](https://github.com/apache/spark/blob/6b156e2fcf9c0c1ed0770a7ad9c54fa374760e17/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala#L834-L842), because `getSD` will be `NULL` for partitioned views. We will get a `NullPointerException`.